src/cpp/common/tls_credentials_options.cc

```diff
@@ -46,10 +47,9 @@ void TlsCredentialsOptions::set_certificate_provider(
 
 void TlsCredentialsOptions::set_crl_provider(
     std::shared_ptr<CrlProvider> crl_provider) {
-  crl_provider_ = std::move(crl_provider);
-  if (crl_provider_ != nullptr) {
+  if (crl_provider != nullptr) {
```

Comment:
Still need to remove this check.  We should unconditionally pass it to the C-core struct, even if it's null.